x-pack/plugins/apm/public/components/app/RumDashboard/URLFilter/URLSearch/index.tsx

```diff
@@ -0,0 +1,201 @@
+
+  const { urlParams, uiFilters } = useUrlParams();
+
+  const { start, end, serviceName } = urlParams;
+  const [searchValue, setSearchValue] = useState('');
+
+  const [forceClosePopover, setForceClosePopover] = useState(false);
+
+  const [debouncedValue, setDebouncedValue] = useState('');
+
+  const [searchRef, setSearchRef] = useState<HTMLInputElement | null>(null);
+
+  const onInputFocus = () => {
+    setForceClosePopover(false);
+  };
+
+  useEffect(() => {
+    if (searchRef) {
+      searchRef.addEventListener('focus', onInputFocus);
+    }
+    return () => {
+      searchRef?.removeEventListener('focus', onInputFocus);
+    };
+  }, [searchRef]);
+
+  useDebounce(
+    () => {
+      setSearchValue(debouncedValue);
+    },
+    250,
+    [debouncedValue]
+  );
+
+  const updateSearchTerm = useCallback(
+    (searchTermN: string) => {
+      const newLocation = {
+        ...history.location,
+        search: fromQuery({
+          ...toQuery(history.location.search),
+          searchTerm: searchTermN,
+        }),
+      };
+      history.push(newLocation);
+    },
+    [history]
+  );
+
+  const [checkedUrls, setCheckedUrls] = useState<string[]>([]);
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (start && end && serviceName) {
+        const { transactionUrl, ...restFilters } = uiFilters;
+
+        return callApmApi({
+          pathname: '/api/apm/rum-client/url-search',
+          params: {
+            query: {
+              start,
+              end,
+              uiFilters: JSON.stringify(restFilters),
+              urlQuery: searchValue,
+            },
+          },
+        });
+      }
+      return Promise.resolve(null);
+    },
+    [start, end, serviceName, uiFilters, searchValue]
+  );
+
+  useEffect(() => {
+    setCheckedUrls(uiFilters.transactionUrl || []);
+  }, [uiFilters]);
+
+  const onChange = (updatedOptions: EuiSelectableTemplateSitewideOption[]) => {
+    const clickedItems = updatedOptions.filter(
+      (option) => option.checked === 'on'
+    );
+
+    setCheckedUrls(clickedItems.map((item) => item.url));
+  };
+
+  // @ts-expect-error
+  const items: EuiSelectableTemplateSitewideOption[] = (data?.items ?? []).map(
+    (item) => ({
+      label: item.url,
+      key: item.url,
+      meta: [
+        {
+          text: I18LABELS.pageViews + ': ' + item.count,
+          type: 'deployment',
+          highlightSearchString: false,
+        },
+        {
+          text: I18LABELS.pageLoadDuration + ': ' + formatToSec(item.pld),
+          type: 'deployment',
+          highlightSearchString: false,
+        },
+      ],
+      url: item.url,
+      checked: checkedUrls?.includes(item.url) ? 'on' : null,
+    })
+  );
+
+  const onInputChange = (e: FormEvent<HTMLInputElement>) => {
+    setDebouncedValue(e.currentTarget.value);
+  };
+
+  const isLoading = status !== 'success';
+
+  const titleText = searchValue
+    ? (data?.total ?? 0) + ' ' + I18LABELS.searchResults
```

Comment:
It might be a better idea to convert `I18LABELS.searchResults` into a function that accepts the `total` and uses `values` to insert it into the translated string, to keep the context with the string being translated.